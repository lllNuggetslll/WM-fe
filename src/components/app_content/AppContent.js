import React, { Fragment } from "react";
import { object, func } from "prop-types";
import ListingCards from "./../listing_cards";
import styled from "styled-components";
import get from "lodash.get";
import { backgroundGray, labelGray } from "./../../constants/colors";

export const AppContent = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: ${backgroundGray};
  padding-bottom: 20px;
`;

export const ListingGroups = styled.div`
  margin-top: 30px;
  h2 {
    text-align: left;
  }
`;

export const Label = styled.div`
  color: ${labelGray};
`;

const regionTypes = ["dispensary", "delivery", "doctor"];
const regionLabels = {
  delivery: {
    label: "Delivery Services",
    icon: "fas fa-car"
  },
  dispensary: {
    label: "Dispensary Storefronts",
    icon: "fas fa-cannabis"
  },
  doctor: {
    label: "Doctors",
    icon: "fas fa-stethoscope"
  }
};

const Content = ({ error, regions, fetchListing }) => {
  const getLabel = (listings, { label, icon }) => {
    if (get(listings, "listings").length) {
      return (
        <Label key={label}>
          <i className={icon} />
          <strong> {label} </strong>
        </Label>
      );
    }
    return <div />;
  };

  return (
    <AppContent>
      {error && <div> {error.message} </div>}
      {regions && (
        <Fragment>
          {regionTypes.map(regionType => (
            <ListingGroups key={regionType}>
              <h2>{getLabel(regions[regionType], regionLabels[regionType])}</h2>
              <ListingCards
                fetchListing={fetchListing}
                listings={get(regions[regionType], "listings")}
              />
            </ListingGroups>
          ))}
        </Fragment>
      )}
    </AppContent>
  );
};

Content.defaultProps = {
  regions: {}
};

Content.propTypes = {
  error: object.isRequired,
  regions: object,
  fetchListing: func.isRequired
};

export default Content;
