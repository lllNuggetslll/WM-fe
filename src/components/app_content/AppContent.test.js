import React from "react";
import { shallow } from "enzyme";

import AppContent from "./AppContent";

describe("AppContent", () => {
  const defaultProps = {
    error: {},
    regions: {
      dispensary: {
        listings: [{}]
      },
      delivery: {
        listings: [{}]
      },
      doctor: {
        listings: [{}]
      }
    },
    fetchListing: () => {}
  };

  it("should render", () => {
    const wrapper = shallow(<AppContent {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
