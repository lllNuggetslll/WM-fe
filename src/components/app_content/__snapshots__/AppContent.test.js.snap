// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContent should render 1`] = `
<styled.div>
  <div>
     
     
  </div>
  <styled.div
    key="dispensary"
  >
    <h2>
      <div
        key="Dispensary Storefronts"
        style={
          Object {
            "color": "#7E7979",
          }
        }
      >
        <i
          className="fas fa-cannabis"
        />
        <strong>
           
          Dispensary Storefronts
           
        </strong>
      </div>
    </h2>
    <withRouter(ListingCards)
      fetchListing={[Function]}
      listings={
        Array [
          Object {},
        ]
      }
    />
  </styled.div>
  <styled.div
    key="delivery"
  >
    <h2>
      <div
        key="Delivery Services"
        style={
          Object {
            "color": "#7E7979",
          }
        }
      >
        <i
          className="fas fa-car"
        />
        <strong>
           
          Delivery Services
           
        </strong>
      </div>
    </h2>
    <withRouter(ListingCards)
      fetchListing={[Function]}
      listings={
        Array [
          Object {},
        ]
      }
    />
  </styled.div>
  <styled.div
    key="doctor"
  >
    <h2>
      <div
        key="Doctors"
        style={
          Object {
            "color": "#7E7979",
          }
        }
      >
        <i
          className="fas fa-stethoscope"
        />
        <strong>
           
          Doctors
           
        </strong>
      </div>
    </h2>
    <withRouter(ListingCards)
      fetchListing={[Function]}
      listings={
        Array [
          Object {},
        ]
      }
    />
  </styled.div>
</styled.div>
`;
